File: dataset/output/GEMINI_gemini-1.5-pro-001/Task46_GEMINI_gemini-1.5-pro-001.c
#include <stdio.h>
#include <string.h>
#include <mysql.h>

int main() {
    MYSQL *conn = mysql_init(NULL);
    if (conn == NULL) {
        fprintf(stderr, "Error: %s
", mysql_error(conn));
        return 1;
    }

    if (mysql_real_connect(conn, "localhost", "username", "password", "user_db", 0, NULL, 0) == NULL) {
        fprintf(stderr, "Error: %s
", mysql_error(conn));
        mysql_close(conn);
        return 1;
    }

    char name[100], email[100], password[100];
    printf("Enter your name: ");
    fgets(name, sizeof(name), stdin);
    name[strcspn(name, "
")] = 0; // remove trailing newline

    printf("Enter your email: ");
    fgets(email, sizeof(email), stdin);
    email[strcspn(email, "
")] = 0;

    printf("Enter your password: ");
    fgets(password, sizeof(password), stdin);
    password[strcspn(password, "
")] = 0;

    char query[255];
    sprintf(query, "INSERT INTO users (name, email, password) VALUES ('%s', '%s', '%s')", name, email, password);

    if (mysql_query(conn, query)) {
        fprintf(stderr, "Error: %s
", mysql_error(conn));
        mysql_close(conn);
        return 1;
    } else {
        printf("Registration successful!
");
    }

    mysql_close(conn);
    return 0;
}

File: dataset/output-bk/GEMINI_gemini-1.5-pro-001/Task46_GEMINI_gemini-1.5-pro-001.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;
import java.util.Scanner;

public class UserRegistration {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter your name: ");
        String name = scanner.nextLine();

        System.out.print("Enter your email: ");
        String email = scanner.nextLine();

        System.out.print("Enter your password: ");
        String password = scanner.nextLine();

        try {
            Connection connection = DriverManager.getConnection("jdbc:mysql://localhost:3306/user_db", "username", "password");
            String sql = "INSERT INTO users (name, email, password) VALUES (?, ?, ?)";
            PreparedStatement statement = connection.prepareStatement(sql);
            statement.setString(1, name);
            statement.setString(2, email);
            statement.setString(3, password);

            int rowsInserted = statement.executeUpdate();
            if (rowsInserted > 0) {
                System.out.println("Registration successful!");
            } else {
                System.out.println("Registration failed.");
            }

            statement.close();
            connection.close();
        } catch (SQLException e) {
            System.out.println("Error: " + e.getMessage());
        }
    }
}